pkg/broker/work_engine.go

```diff
@@ -50,10 +54,65 @@ func (engine *WorkEngine) StartNewAsyncJob(
 	if token == "" {
 		token = engine.Token()
 	}
-	go work.Run(token, engine.topic(topic))
+	go engine.startJob(token, work, topic)
```

Comment:
`startJob` is better and more clear.

Reply:
:+1: I'm not sure it should be a private fn though? I suppose it's fine since we're really the only ones using it but I thought it was useful to design it from a perspective of "this is a public lib". No requested changes, but spitballing.